src/Data/DataSearchResult/DataSearchResultDataClass.ts

```diff
@@ -0,0 +1,85 @@
+import { currentLanguage, load, provideDataClass } from 'scrivito'
+import { pisaClient } from '../pisaClient'
+import { getDataItemUrl } from './getDataItemUrl'
+import { dataSearchResultIndexFallback } from './dataSearchResultIndexFallback'
+
+export const DataSearchResult = provideDataClass('DataSearchResult', {
+  title: async () =>
+    (await load(() => currentLanguage())) === 'de'
+      ? 'Suchtreffer (Daten)'
+      : 'Search result (data)',
+  attributes: async () => {
+    const lang = await load(() => currentLanguage())
+
+    return {
+      entity: ['string', { title: lang === 'de' ? 'Entität' : 'Entity' }],
+      snippet: ['string', { title: lang === 'de' ? 'Schnipsel' : 'Snippet' }],
+      title: ['string', { title: lang === 'de' ? 'Titel' : 'Title' }],
+      url: ['unknown', { title: 'URL' }],
+    }
+  },
+  connection: {
+    async index(params) {
+      if (Object.keys(params.filters()).length > 0) {
+        throw new Error('Search result (data) does not support filtering.')
+      }
+
+      if (params.order().length > 0) {
+        throw new Error('Search result (data) does not support sorting.')
+      }
+
+      const globalResultClient = await pisaClient('global-result')
+      if (!globalResultClient) return dataSearchResultIndexFallback(params)
+
+      if (!params.search()) return { results: [], count: 0 }
+
+      const classNames = [
+        'Contract',
+        'Document',
+        'Event',
+        'Order',
+        'Quote',
+        'ServiceObject',
+        'Ticket',
+      ]
+
+      const {
+        results: rawResults,
+        count,
+        continuation,
+      } = (await globalResultClient.get('', {
+        params: {
+          _continuation: params.continuation(),
+          _count: params.includeCount().toString(),
+          _limit: params.limit().toString(),
+          _search: params.search(),
+          entity: classNames.join(','),
+        },
+      })) as {
+        results: Array<{
+          _id: string
+          entity: string
+          matches: Array<[number, number]>
+          snippet: string
+          title: string
+        }>
+        count?: number
+        continuation?: string
+      }
+
+      const results = await Promise.all(
+        rawResults.map(async ({ _id, entity, title, snippet }) => {
+          return {
+            _id,
+            entity,
+            snippet,
+            title,
+            url: await getDataItemUrl(entity, _id),
```

Comment:
I am sure that there must be a computation benefit from calculating the URLs in parallel, however I I am not sure if that benefit is worth the hassle of all the `load`s and `await`s. I would propose this simplification instead:
```ts
const results = await load(() =>
  rawResults.map(({ _id, entity, ...rest }) => ({
    _id,
    entity,
    ...rest,
    url: getDataItemUrl(entity, _id),
  }))
);
```

This would also eliminate the need of a `load` inside `getDataItemUrl`.
But up to you.

Reply:
I prefer the parallel version.